### PR TITLE
Action to publish packages to mzio

### PR DESCRIPTION
**Pull request by SteffenHeu**
Could not figure out yet how to publish the convention plugins as they are not included in the standard build logic.

### CODE REVIEW

File: .github/workflows/push_stable_packages.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          java-version: '22.0.1'
+          distribution: 'temurin'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@af1da67850ed9a4cedd57bfd976089dd991e2582 # v4.0.0
```
Comment (robinschmid) [inline .github/workflows/push_stable_packages.yml:27]: just uses: gradle/actions/setup-gradle@v4?

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                        url = uri("https://maven.pkg.github.com/mzio-gmbh/mzio_mzmine")
+                        credentials {
+                            username = System.getenv("PUBLISH_PACKAGE_USERNAME")
+                            password = System.getenv("PUBLISH_PACKAGE_TOKEN")
+                        }
+                    }
+                } else {
+                    maven {
+                        url = uri(layout.projectDirectory.dir("../local-repo/"))
+                    }
```
Comment (robinschmid) [inline build.gradle.kts:35]: Is the local repo needed for the convention pugins to be always shared through artifactory?
Or does this also work without publishing it to local?

File: settings.gradle.kts [hunk 3]
```diff
@@ -20,4 +20,4 @@ include(
     "javafx-framework",
     "config",
 )
-//includeBuild("convention-plugins")
\ No newline at end of file
+//includeBuild("convention-plugins")
```
Comment (robinschmid) [inline settings.gradle.kts:23]: so it seems this is not needed? Did you try if this links up the convention plugins or does it break something?

**Review by robinschmid — COMMENTED** [review]
Looks good. Now just a question if it works to access the packages

File: .github/workflows/push_stable_packages.yml (hunk 1 above)
Comment (SteffenHeu) [reply .github/workflows/push_stable_packages.yml:27]:
```
# This workflow uses actions that are not certified by GitHub.
# They are provided by a third-party and are governed by
# separate terms of service, privacy policy, and support
# documentation.

# GitHub recommends pinning actions to a commit SHA.
# To get a newer version, you will need to update the SHA.
# You can also reference a tag or branch, but the action may change without warning.
```

took the workflow from the github docs, had this comment

File: settings.gradle.kts (hunk 3 above)
Comment (SteffenHeu) [reply settings.gradle.kts:23]: i does not cause the plugins to be published on gradlew publish

File: build.gradle.kts (hunk 2 above)
Comment (SteffenHeu) [reply build.gradle.kts:35]: i just thought it would be nice to be able to test the build locally as well in the future. that's why i added this.

**Review by robinschmid — APPROVED** [review]
only some comments

**Pull request merged**